Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+public sealed class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private CharacterController characterController;
+    [SerializeField] private float movingSpeed;
+
+
+    private Vector2 _moveDirection;
+    private float _horizontal;
+    private float _vertical;
+
+    private void Update()

```

Comment:
Probably want to do input in the fixed update. You don't want people being able to press keys or not to be frame-dependent.